inc/Engine/Admin/Settings/Page.php

```diff
@@ -2251,4 +2214,52 @@ public function display_radio_options_sub_fields( $sub_fields ) {
 		$sub_fields = $this->settings->set_radio_buttons_sub_fields_value( $sub_fields );
 		$this->render->render_fields( $sub_fields );
 	}
+
+	/**
+	 * Render mobile cache option.
+	 *
+	 * @return void
+	 */
+	public function display_mobile_cache_option() : void {
+		if ( (bool) get_rocket_option( 'cache_mobile', 0 ) ) {
+			return;
+		}
+
+		$data = $this->beacon->get_suggest( 'mobile_cache' );
+		echo $this->generate( 'settings/mobile-cache', $data ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Dynamic content is properly escaped in the view.
+	}
+
+	/**
+	 * Callback method for the AJAX request to mobile cache.
+	 *
+	 * @return void
+	 */
+	public function enable_mobile_cache() : void {
+		check_ajax_referer( 'rocket-ajax', 'nonce', true );
+
+		if ( ! current_user_can( 'rocket_manage_options' ) ) {
+			wp_send_json_error();
+			return;
+		}
+
+		$this->options->set( 'cache_mobile', 1 );
+		$this->options->set( 'do_caching_mobile_files', 1 );
+		update_option( rocket_get_constant( 'WP_ROCKET_SLUG', 'wp_rocket_settings' ), $this->options->get_options() );
+
+		wp_send_json_success();
+	}
+
+	/**
+	 * Enable Separate cache files option on upgrade.
+	 *
+	 * @return void
+	 */
+	public function enable_separate_cache_files_mobile() : void {
+		if ( ! (bool) get_rocket_option( 'cache_mobile', 0 ) ) {
+			return;
+		}
+
+		$this->options->set( 'do_caching_mobile_files', 1 );
+		update_option( rocket_get_constant( 'WP_ROCKET_SLUG', 'wp_rocket_settings' ), $this->options->get_options() );
+	}
```

Comment:
Missing a check for the current version number vs the update version number.

We can also bail-out early if the option is already enabled